Engine/TransactionHandlers/BrokerageTransactionHandler.cs

```diff
@@ -820,11 +818,16 @@ private void HandleOrderEvent(OrderEvent fill)
             {
                 Log.Debug("BrokerageTransactionHandler.HandleOrderEvent(): " + fill);
                 Interlocked.Exchange(ref _lastFillTimeTicks, DateTime.Now.Ticks);
+
+                // set the price currency on the fill and on the order
+                var security = _algorithm.Securities[fill.Symbol];
+                fill.FillPriceCurrency = security.SymbolProperties.QuoteCurrency;
+                order.PriceCurrency = fill.FillPriceCurrency;
```

Comment:
I think that's a good plan. We'll ensure that all orders are tagged with a currency when we create the order and if we find a mismatch we can `Log.Error` it and continue. For now we can assume the brokerage populates the order event with the price in proper currency units and use the order event ctor to set the currency from the order object